tools/coqdep_common.ml

```diff
@@ -280,6 +285,24 @@ let string_of_dependency_list suffix_for_require deps =
     in
   String.concat " " (List.map string_of_dep deps)
 
+let rec phys_path_best_match (prefix: string list) (logpath: string list) :  (string (* physical path corresponding to a prefix of logpath *) * string list (* unmatched suffix *)) option =
+  match logpath with
+  | [] -> (match find_physpath prefix with
+          | None -> None
+          | Some p -> Some (p,[]))
+  | h::tl ->
+          match phys_path_best_match (prefix@[h]) tl with
+          | None -> (match find_physpath prefix with
+                    | None -> None
+                    | Some p -> Some (p,h::tl)
+                    )
+          | Some p -> Some p
+
+let phys_path (logpath: string list) : string option =
+  match phys_path_best_match [] logpath with
```

Comment:
Option.map could be helpful here.

Reply:
With that, I get the following error: 
````
make[1]: Entering directory '/home/abhishek/work/coq'
OCAMLBEST -o bin/coqdep_boot
File "_none_", line 1:
Error: No implementations provided for the following modules:
         Option referenced from tools/coqdep_common.cmx
````